federation/pkg/federatedtypes/scheduling.go

```diff
@@ -113,6 +170,23 @@ func (a *replicaSchedulingAdapter) GetSchedule(obj pkgruntime.Object, key string
 		}
 		return clientset.Core().Pods(metadata.GetNamespace()).List(metav1.ListOptions{LabelSelector: selector.String()})
 	}
+
+	initializedState, hpaControlled, err := initializeScheduleState(obj, clusterNames)
+	if err != nil {
+		return nil, err
+	}
+
+	if hpaControlled {
+		state, err := simpleSchedule(initializedState, key, objectGetter)
+		if err != nil {
+			return nil, err
+		}
+		return &ReplicaSchedulingInfo{
+			ScheduleState: state,
+			Status:        ReplicaSchedulingStatus{},
```

Comment:
I think this was confusing because of the naming of the two structures also. I have updated the name with comments explaining the difference to avoid confusion [also explained in this comment](https://github.com/kubernetes/kubernetes/pull/49950/files#r1332031210).